src/components/ui/multi-select.tsx

```diff
@@ -34,201 +39,203 @@ interface MultiSelectProps extends React.ButtonHTMLAttributes<HTMLButtonElement>
   onValueChange: (value: string[]) => void;
   value: string[];
   placeholder: string;
-  modalPopover?: boolean;
-  asChild?: boolean;
   className?: string;
+  selectionSummary?: string;
+  translationBasekey?: string;
 }
 
 export function MultiSelect({
   options,
   onValueChange,
   value = [],
   placeholder,
-  modalPopover = false,
   className,
   ref,
+  selectionSummary,
+  translationBasekey,
   ...props
-}: React.ComponentProps<"button"> & MultiSelectProps) {
+}: ButtonProps & MultiSelectProps) {
   const [selectedValues, setSelectedValues] = React.useState<string[]>(value);
   const [open, setOpen] = React.useState(false);
   const isMobile = useBreakpoints({ default: true, sm: false });
 
   React.useEffect(() => {
     setSelectedValues(value);
-  }, [value]);
+  }, [value, open]);
+  React.useEffect(() => {
+    if (open == false) onValueChange(selectedValues);
+  }, [open]);
 
   const { t } = useTranslation();
 
-  const toggleOption = (option: string) => {
-    const newSelectedValues = selectedValues.includes(option)
-      ? selectedValues.filter((value) => value !== option)
-      : [...selectedValues, option];
-    setSelectedValues(newSelectedValues);
-    onValueChange(newSelectedValues);
-  };
-
-  const handleClear = () => {
-    setSelectedValues([]);
-    onValueChange([]);
-  };
-
-  const handleToggle = () => {
-    setOpen((prev) => !prev);
-  };
-
-  const toggleAll = () => {
-    if (selectedValues.length === options.length) {
-      handleClear();
-    } else {
-      const allValues = options.map((option) => option.value);
-      setSelectedValues(allValues);
-      onValueChange(allValues);
-    }
-  };
-
-  const handleTouchStart = (e: React.TouchEvent) => {
-    e.stopPropagation();
+  const handleToggleOption = (option: string) => {
+    setSelectedValues((prevSelectedValues) =>
+      prevSelectedValues.includes(option)
+        ? prevSelectedValues.filter((v) => v !== option)
+        : [...prevSelectedValues, option],
+    );
   };
-
-  const handleTouchMove = (e: React.TouchEvent) => {
-    e.stopPropagation();
+  const handleSelectAll = () => {
+    setSelectedValues((prevSelectedValues) => {
+      if (prevSelectedValues.length === options.length) return [];
+      return options.map((o) => o.value);
+    });
   };
 
   const triggerButton = (
```

Comment:
go ahead and convert these to child components instead. We have a few cases where we are defining logic like this, shouldn't be doing that as it's not optimal.

Either it should be a child component or inline, not set to variables.